### DISCUSSION TITLE
question about usage in chat apps

### DISCUSSION THREAD

**Discussion by Ahmed-Elrayes** [sachinchoolur/lightGallery#1123] [Q&A] [answered]
Hello, i've a question.
at first i wanna say i'm not so good with frontend stuff, but i'm doing a little chat app for admin panel to only see that chat messages realtime and the chat might have images, videos ..etc 
is it possible to make light gallery so dynamic like if someone sent a picture after page loads it will add the newly added image to the lightGallery dynamically so when i click on it it opens normally? without destroying and reinitializing lightGallery every time it loads an image?

**Comment by sachinchoolur** [comment, marked as answer]
Hey @Ahmed-Elrayes,

Sorry for the delayed response, I forgot to check the discussion section.

You can just need to call the `refresh()` method once a new photo or video is added
Docs - https://www.lightgalleryjs.com/docs/methods/#refresh
Demo - https://www.lightgalleryjs.com/demos/infinite-scrolling/


If you want to make it more dynamic, you can try using dynamic mode and refresh it, and open using `openGallery` when the user clicks on any image

**Reply by Ahmed-Elrayes** [reply]
> Hey @Ahmed-Elrayes,
> 
> Sorry for the delayed response, I forgot to check the discussion section.
> 
> You can just need to call the `refresh()` method once a new photo or video is added
> Docs - https://www.lightgalleryjs.com/docs/methods/#refresh
> Demo - https://www.lightgalleryjs.com/demos/infinite-scrolling/
> 
> If you want to make it more dynamic, you can try using dynamic mode and refresh it, and open using `openGallery` when the user clicks on any image


that's so great, thanks a lot.
i have a little question, if i have a page with paginating, and i did move to another page, will the refresh method work there too? or i will have to destroy the previous Light Gallery and create a new one?

PS: there will be multiple groups ..etc 
as example this will be used in posts container so for each post will be a separate group of pictures, videos ..etc

**Reply by sachinchoolur** [reply]
As long as the container exists in the dom, refresh method should work fine. If you want, you can set any parent element as container and select image/videos using lightGallery `selector` setting

Docs - https://www.lightgalleryjs.com/demos/html-markup/
Demo - https://www.lightgalleryjs.com/demos/medium-zoom/

let me know if you have any questions

**Reply by Ahmed-Elrayes** [reply]
oh, that's great then, i will try it out and tell you, ty
